### PR TITLE
do_cmake: warn users about slow debug builds

### PR DESCRIPTION
**Pull request by markhpc**
Spent more time than I want to admit bisecting master over the weekend looking for a 5X random write performance regression on NVMe.  Finally I figured out that do_cmake.sh was changed to create debug builds by default in #24133. :sweat_smile:  This PR adds copious warnings so that I hopefully won't have to bisect this again in six months after I've forgotten that we changed the default behavior.

- [ ] References tracker ticket
- [X] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,10 @@ Build instructions:
 	cd build
 	make
 
+(Note: do_cmake.sh now defaults to creating a debug build of ceph that can
+be up to 5x slower with some workloads. Please pass 
+"-DCMAKE_BUILD_TYPE=Release" to do_cmake.sh to create a non-debug release.)
```
Comment (tchaikov) [inline README.md]: i'd suggest s/Release/RelWithDebInfo/ in this changeset to ensure that user will have debug symbol generated.

Comment (liewegas) [reply README.md]: +1

Comment (liewegas) [reply README.md]: I'd also modify do_cmake.sh itself to print this out in a loud/noticable way at the top or bottom
```
****
  WARNING: debug build selected -- performance will be bad but gdb will be happy
****
```
or something

File: do_cmake.sh [hunk 2]
```diff
@@ -48,3 +48,4 @@ erasure code dir = lib
 EOF
 
 echo done.
+echo "\n****\nWARNING: do_cmake.sh now creates debug builds by default. Performance\nmay be severely affected. Please use -DCMAKE_BUILD_TYPE=RelWithDebInfo\nif a performance sensitive build is required.\n****"
```
Comment (tchaikov) [inline do_cmake.sh]: could use here document for better formatting.

**Pull request merged**